core/src/logger.ts

```diff
@@ -0,0 +1,196 @@
+  const orgConsole = (isIos ? {} : gbl.console) as any;
+
+  const useFallbackLogging =
+    !!gbl.console && Object.keys(gbl.console).length === 0;
+  if (useFallbackLogging && gbl.console) {
+    gbl.console.warn('Advance console logging disabled.');
+  }
+
+  // list log functions bridged to native log
+  const bridgedLevels: { [key: string]: boolean } = {
+    debug: true,
+    error: true,
+    info: true,
+    log: true,
+    trace: true,
+    warn: true,
+  };
+
+  if (isIos && gbl.console) {
+    Object.keys(gbl.console).forEach(level => {
+      if (typeof gbl.console[level] === 'function') {
+        // loop through all the console functions and keep references to the original
+        orgConsole[level] = gbl.console[level];
+        gbl.console[level] = function capacitorConsole() {
+          let msgs: any[] = Array.prototype.slice.call(arguments);
+
+          // console log to browser
+          orgConsole[level].apply(gbl.console, msgs);
+
+          if (bridgedLevels[level]) {
+            // send log to native to print
+            try {
+              // convert all args to strings
+              msgs = msgs.map(arg => {
+                if (typeof arg === 'object') {
+                  try {
+                    arg = JSON.stringify(arg);
+                  } catch (e) {}
+                }
+                // convert to string
+                return String(arg);
+              });
+              instance.toNative('Console', 'log', {
+                level: level,
+                message: msgs.join(' '),
+              });
+            } catch (e) {
+              // error converting/posting console messages
+              orgConsole.error.apply(gbl.console, e);
+            }
+          }
+        };
+      }
+    });
+  }
+
+  instance.logJs = (message, level) => {
+    switch (level) {
+      case 'error':
+        gbl.console.error(message);
+        break;
+      case 'warn':
+        gbl.console.warn(message);
+        break;
+      case 'info':
+        gbl.console.info(message);
+        break;
+      default:
+        gbl.console.log(message);
+    }
+  };
+
+  instance.handleError = (e: Error) => gbl.console.error(e);
+
+  instance.handleWindowError = (msg, url, lineNo, columnNo, error) => {
+    const str = msg.toLowerCase();
+    const substring = 'script error';
+
+    if (str.indexOf(substring) > -1) {
+      // Some IE issue?
+    } else {
+      const errObj = {
+        type: 'js.error',
+        error: {
+          message: msg,
+          url: url,
+          line: lineNo,
+          col: columnNo,
+          errorObject: JSON.stringify(error),
+        },
+      };
+
+      if (error !== null) {
+        instance.handleError(error);
+      }
+
+      if (postToNative) {
+        postToNative(errObj);
+      }
+    }
+
+    return false;
+  };
+
+  if (instance.DEBUG) {
+    window.onerror = instance.handleWindowError;
+  }
+
+  instance.logToNative = (call: CallData) => {
+    if (!useFallbackLogging) {
```

Comment:
Shouldn't this logging only be done if `instance.DEBUG`? There's a few reasons why:

- We really don't want to see this info most of the time. If we are doing our own logging, the spam from native calls obscures our own logs.

- There is a very real cost at runtime. You have to go through the JS<->Native bridge code, then the bridge has to parse the call and execute a dynamic dispatch.